### PR TITLE
[TS migration] Migrate 'SettingsProfileDetails' page to TypeScript

### PR DESCRIPTION
**Pull request by kubabutkiewicz**
### Fixed Issues

$ https://github.com/Expensify/App/issues/25214

### Tests

1. Login into app
2. Go to Settings  > Profile
3. Verify that Legal name page is working
4. Verify that Date of Birth Page is working
5. Verify that Address Page is working

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Login into app
2. Go to Settings > Profile > Personal Details 
3. Verify that Legal name page is working
4. Verify that Date of Birth Page is working
5. Verify that Address Page is working

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/25230417/92a963f4-a975-4983-81ec-0a75bd48c6c8

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/25230417/fb0cba55-ae27-41b6-98f4-da2dd0dcfe71

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/25230417/016aad85-1c5b-429d-8ac1-2f19e097c861

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/25230417/b061c05c-d252-4851-a8bd-223f84019dec

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/25230417/9268e57a-160e-4a45-b922-035b6501e386

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/25230417/66b913c0-5341-44f4-94af-bc4ba881bd27

</details>

### CODE REVIEW

File: src/pages/settings/Profile/PersonalDetails/LegalNamePage.tsx [hunk 1]
```diff
@@ -1,82 +1,70 @@
...
 
-        if (!ValidationUtils.isValidLegalName(values.legalFirstName)) {
+        if (!ValidationUtils.isValidLegalName(values.legalFirstName ?? '')) {
             ErrorUtils.addErrorMessage(errors, 'legalFirstName', 'privatePersonalDetails.error.hasInvalidCharacter');
-        } else if (_.isEmpty(values.legalFirstName)) {
+        } else if (values.legalFirstName === '') {
             errors.legalFirstName = 'common.error.fieldRequired';
         }
-        if (ValidationUtils.doesContainReservedWord(values.legalFirstName, CONST.DISPLAY_NAME.RESERVED_NAMES)) {
+        if (ValidationUtils.doesContainReservedWord(values.legalFirstName ?? '', CONST.DISPLAY_NAME.RESERVED_NAMES as unknown as string[])) {
```
Comment (VickyStash) [inline src/pages/settings/Profile/PersonalDetails/LegalNamePage.tsx:51]: I've checked and this function is used only with this value `CONST.DISPLAY_NAME.RESERVED_NAMES`, so maybe we can update the typing or just use by default, not sure...
```suggestion
        if (ValidationUtils.doesContainReservedWord(values.legalFirstName ?? '', CONST.DISPLAY_NAME.RESERVED_NAMES)) {
```

File: src/pages/settings/Profile/PersonalDetails/PersonalDetailsInitialPage.tsx [hunk 2]
```diff
@@ -1,72 +1,52 @@
...
-    ...withLocalizePropTypes,
+    privatePersonalDetails: OnyxEntry<PrivatePersonalDetails>;
 };
 
-const defaultProps = {
-    privatePersonalDetails: {
+type PersonalDetailsInitialPageProps = PersonalDetailsInitialPageOnyxProps;
+
+function PersonalDetailsInitialPage({
+    privatePersonalDetails = {
```
[the remark this answers is not in the document]
Comment (kubabutkiewicz) [reply src/pages/settings/Profile/PersonalDetails/PersonalDetailsInitialPage.tsx:27]: yeah you are right I will remove that default value

File: src/libs/ValidationUtils.ts [hunk 3]
```diff
@@ -176,7 +176,7 @@ function meetsMaximumAgeRequirement(date: string): boolean {
 /**
  * Validate that given date is in a specified range of years before now.
  */
-function getAgeRequirementError(date: string, minimumAge: number, maximumAge: number): string | Array<string | Record<string, string>> {
+function getAgeRequirementError(date: string, minimumAge: number, maximumAge: number): string | [string, Record<string, string>] {
```
Comment (fabioh8010) [inline src/libs/ValidationUtils.ts:179]:
```suggestion
function getAgeRequirementError(date: string, minimumAge: number, maximumAge: number): Localize.MaybePhraseKey {
```

this works?

File: src/pages/settings/Profile/PersonalDetails/AddressPage.tsx [hunk 4]
```diff
@@ -10,67 +10,48 @@ import useLocalize from '@hooks/useLocalize';
...
+        values.addressLine2?.trim() ?? '',
+        values.city.trim(),
+        values.state.trim(),
+        values?.zipPostCode?.trim().toUpperCase() ?? '',
+        values.country,
+    );
 }
 
-function AddressPage({privatePersonalDetails, route}) {
+function AddressPage({privatePersonalDetails = {address: {street: '', city: '', state: '', zip: '', country: ''}}, route}: AddressPageProps) {
```
Comment (fabioh8010) [inline src/pages/settings/Profile/PersonalDetails/AddressPage.tsx:43]: Is possible to get rid of the default value?

File: src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx [hunk 5]
```diff
@@ -55,19 +40,18 @@ function CountrySelectionPage({route, navigation}) {
...
-
+        (option: CountryData) => {
+            const backTo = route.params.backTo ?? '';
             // Check the navigation state and "backTo" parameter to decide navigation behavior
-            if (navigation.getState().routes.length === 1 && _.isEmpty(backTo)) {
+            if (navigation.getState().routes.length === 1 && !backTo) {
                 // If there is only one route and "backTo" is empty, go back in navigation
                 Navigation.goBack();
-            } else if (!_.isEmpty(backTo) && navigation.getState().routes.length === 1) {
+            } else if (backTo !== '' && navigation.getState().routes.length === 1) {
```
Comment (fabioh8010) [inline src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx:49]:
```suggestion
            } else if (!!backTo && navigation.getState().routes.length === 1) {
```

This works?

File: src/pages/settings/Profile/PersonalDetails/DateOfBirthPage.tsx [hunk 6]
```diff
@@ -1,59 +1,47 @@
...
-const defaultProps = {
-    privatePersonalDetails: {
-        dob: '',
-    },
+    privatePersonalDetails: OnyxEntry<PrivatePersonalDetails>;
 };
+type DateOfBirthPageProps = DateOfBirthPageOnyxProps;
 
-function DateOfBirthPage({translate, privatePersonalDetails}) {
+function DateOfBirthPage({privatePersonalDetails = {dob: ''}}: DateOfBirthPageProps) {
```
Comment (fabioh8010) [inline src/pages/settings/Profile/PersonalDetails/DateOfBirthPage.tsx:29]: Same, is possible to get rid of the default value?

File: src/pages/settings/Profile/PersonalDetails/LegalNamePage.tsx [hunk 7]
```diff
@@ -1,82 +1,71 @@
...
-    const validate = useCallback((values) => {
-        const errors = {};
+    const validate = useCallback((values: OnyxFormValuesFields<typeof ONYXKEYS.FORMS.LEGAL_NAME_FORM>) => {
+        const errors: Errors = {};
 
-        if (!ValidationUtils.isValidLegalName(values.legalFirstName)) {
+        if (!ValidationUtils.isValidLegalName(values.legalFirstName ?? '')) {
             ErrorUtils.addErrorMessage(errors, 'legalFirstName', 'privatePersonalDetails.error.hasInvalidCharacter');
-        } else if (_.isEmpty(values.legalFirstName)) {
+        } else if (values.legalFirstName === '') {
```
Comment (fabioh8010) [inline src/pages/settings/Profile/PersonalDetails/LegalNamePage.tsx:49]:
```suggestion
        } else if (!values.legalFirstName) {
```

WDYT?

File: src/pages/settings/Profile/PersonalDetails/LegalNamePage.tsx [hunk 8]
```diff
@@ -1,82 +1,71 @@
...
-            ErrorUtils.addErrorMessage(errors, 'legalFirstName', ['common.error.characterLimitExceedCounter', {length: values.legalFirstName.length, limit: CONST.LEGAL_NAME.MAX_LENGTH}]);
+        if ((values.legalFirstName?.length ?? 0) > CONST.LEGAL_NAME.MAX_LENGTH) {
+            ErrorUtils.addErrorMessage(errors, 'legalFirstName', ['common.error.characterLimitExceedCounter', {length: values.legalFirstName?.length, limit: CONST.LEGAL_NAME.MAX_LENGTH}]);
         }
 
-        if (!ValidationUtils.isValidLegalName(values.legalLastName)) {
+        if (!ValidationUtils.isValidLegalName(values.legalLastName ?? '')) {
             ErrorUtils.addErrorMessage(errors, 'legalLastName', 'privatePersonalDetails.error.hasInvalidCharacter');
-        } else if (_.isEmpty(values.legalLastName)) {
+        } else if (values.legalLastName === '') {
```
Comment (fabioh8010) [inline src/pages/settings/Profile/PersonalDetails/LegalNamePage.tsx:61]: Same

File: src/types/onyx/OnyxCommon.ts [hunk 9]
```diff
@@ -8,7 +9,9 @@ type PendingFields<TKey extends string = string> = Record<TKey, PendingAction |
 
 type ErrorFields<TKey extends string = string> = Record<TKey, Errors | null | undefined>;
 
-type Errors = Record<string, string | null>;
+type Error = string | MaybePhraseKey;
+
+type Errors = Record<string, Error | null>;
```
Comment (fabioh8010) [inline src/types/onyx/OnyxCommon.ts:14]: What's the reason of this change? Let's be careful because `Errors` is a type used in many files

Comment (kubabutkiewicz) [reply src/types/onyx/OnyxCommon.ts:14]: Its because for example of this https://github.com/Expensify/App/blob/f697c897d05588f58aba426199f4cc08638711b3/src/pages/settings/Profile/PersonalDetails/LegalNamePage.js#L65-L67

here we are not adding just string but this `MaybePhraseKey`

**Review by fabioh8010 — APPROVED** [review]

**Review by blazejkustra — APPROVED** [review]
LGTM

**Comment by situchan** [comment]
@kubabutkiewicz please fix conflict

File: src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx [hunk 10]
```diff
@@ -1,45 +1,30 @@
...
-        getState: PropTypes.func.isRequired,
-    }).isRequired,
-};
-
-function CountrySelectionPage({route, navigation}) {
+function CountrySelectionPage({route, navigation}: CountrySelectionPageProps) {
     const [searchValue, setSearchValue] = useState('');
     const {translate} = useLocalize();
-    const currentCountry = lodashGet(route, 'params.country');
+    const currentCountry = route.params.country;
```
Comment (situchan) [inline src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx:22]: Shouldn't this be `route.params?.country`? This is inconsistent with https://github.com/Expensify/App/pull/35307/files#diff-02dc5171dfb0335e880ff4994705c5607f8f7a82de55bd1d4fa241c499d9232dR47

File: src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx [hunk 11]
```diff
@@ -55,19 +40,18 @@ function CountrySelectionPage({route, navigation}) {
     const headerMessage = searchValue.trim() && !searchResults.length ? translate('common.noResultsFound') : '';
 
     const selectCountry = useCallback(
-        (option) => {
-            const backTo = lodashGet(route, 'params.backTo', '');
-
+        (option: CountryData) => {
+            const backTo = route.params.backTo ?? '';
```
Comment (situchan) [inline src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx:44]: Same here. `route.params` always exists?

File: src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx [hunk 12]
```diff
@@ -82,9 +66,9 @@ function CountrySelectionPage({route, navigation}) {
                 title={translate('common.country')}
                 shouldShowBackButton
                 onBackButtonPress={() => {
-                    const backTo = lodashGet(route, 'params.backTo', '');
-                    const backToRoute = backTo ? `${backTo}?country=${currentCountry}` : '';
-                    Navigation.goBack(backToRoute);
+                    const backTo = route.params.backTo ?? '';
```
Comment (situchan) [inline src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx:69]: Same here ^

File: src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx [hunk 13]
```diff
@@ -82,9 +66,9 @@ function CountrySelectionPage({route, navigation}) {
...
+                    const backToRoute = backTo ? (`${backTo}?country=${currentCountry}` as const) : undefined;
```
Comment (situchan) [inline src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx:70]: Why is this updated to `undefined`? Originally was empty string.

File: src/libs/Navigation/types.ts [hunk 14]
```diff
@@ -90,8 +90,13 @@ type SettingsNavigatorParamList = {
     [SCREENS.SETTINGS.PROFILE.TIMEZONE_SELECT]: undefined;
     [SCREENS.SETTINGS.PROFILE.LEGAL_NAME]: undefined;
     [SCREENS.SETTINGS.PROFILE.DATE_OF_BIRTH]: undefined;
-    [SCREENS.SETTINGS.PROFILE.ADDRESS]: undefined;
-    [SCREENS.SETTINGS.PROFILE.ADDRESS_COUNTRY]: undefined;
+    [SCREENS.SETTINGS.PROFILE.ADDRESS]: {
+        country: string;
```
[the remark this answers is not in the document]
Comment (kubabutkiewicz) [reply src/libs/Navigation/types.ts:94]: No its not will adjust that

File: src/pages/settings/Profile/PersonalDetails/AddressPage.tsx [hunk 15]
```diff
@@ -10,66 +10,47 @@ import useLocalize from '@hooks/useLocalize';
...
  */
-function updateAddress(values) {
-    PersonalDetails.updateAddress(values.addressLine1.trim(), values.addressLine2.trim(), values.city.trim(), values.state.trim(), values.zipPostCode.trim().toUpperCase(), values.country);
+function updateAddress(values: Address) {
+    PersonalDetails.updateAddress(
+        values.addressLine1?.trim() ?? '',
+        values.addressLine2?.trim() ?? '',
+        values.city.trim(),
+        values.state.trim(),
+        values?.zipPostCode?.trim().toUpperCase() ?? '',
```
[the remark this answers is not in the document]
Comment (kubabutkiewicz) [reply src/pages/settings/Profile/PersonalDetails/AddressPage.tsx:37]: Actually here , only `addressLine2` is an optional field on the address so I adjust that tat this field is the only optional one

File: src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx (hunk 10 above)
Comment (kubabutkiewicz) [reply src/pages/settings/Profile/PersonalDetails/CountrySelectionPage.tsx:22]: Actually I think params are always there 
image
So this should be fine to left it as it is

**Comment by situchan** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by situchan** [comment]
Please also update QA Steps

> 2. Go to Settings > Profile > Personal Details

`Personal Details` page is deprecated. Now all sub-pages are directly in Profile page

**Review by situchan — APPROVED** [review]
:shipit:

**Review by situchan — APPROVED** [review]

**Comment by situchan** [comment]
@kubabutkiewicz there's conflict

**Comment by kubabutkiewicz** [comment]
@situchan conflicts resolved 😄

**Comment by situchan** [comment]
@tylerkaraszewski all yours

**Comment by tylerkaraszewski** [comment]
There are conflicts again.

**Comment by kubabutkiewicz** [comment]
will take a look at this today

**Review by tylerkaraszewski — APPROVED** [review]

**Comment by tylerkaraszewski** [comment]
I merged this, but it caused the failure here:
https://github.com/Expensify/App/issues/37215

I am going to revert this PR and let you re-create it kubabutkiewicz, sorry about the extra work, but you can look at the failure above and resolve that in the next iteration.

**Comment by kubabutkiewicz** [comment]
will take a look soon on that
